search-relevancy-test/src/search_relevancy_test/result_logger.clj

```diff
@@ -0,0 +1,110 @@
+(ns search-relevancy-test.result-logger
+  "Functions for writing test results to file"
+  (:require
+   [clj-time.core :as time]
+   [clj-time.format :as time-format]
+   [clojure.data.csv :as csv]
+   [clojure.java.io :as io]
+   [clojure.string :as string]
+   [search-relevancy-test.core :as core]
+   [search-relevancy-test.reporter :as reporter]))
+
+(def local-test-run-csv
+  "local_test_runs.csv")
+
+(def test-run-history-csv
+  "test_run_history.csv")
+
+(defn- get-last-run
+  "Read the CSV file and get the data from the last run, if exists"
+  [csv-file]
+  (when-let [last-run (->> (core/read-csv local-test-run-csv)
+                           (map #(assoc % :date (time-format/parse (:date %))))
+                           (group-by :date)
+                           sort
+                           last)]
+    (val last-run)))
+
+(defn- result-summary->csv-row
+  "Create the CSV row data from the result summary"
+  [result-summary]
+  {:anomaly "SUMMARY"
+   :test ""
+   :description ""
+   :pass (:pass result-summary)
+   :discounted-cumulative-gain (:average-dcg result-summary)
+   :reciprocal-rank (:mean-reciprocal-rank result-summary)
+   :result-description (:result-description result-summary)})
+
+(defn- changed-from-last-run
+  "Determine if for the test and anomaly the result has changed from the last run.
+  If so, return a string with differences from the last run. If not, return nil."
+  [result last-run]
+  (when-let [last-run-result (some #(when (and (= (:anomaly %) (:anomaly result))
```

Comment:
This is a pretty big anonymous function. It might be worth naming it so it gets a docstring (and maybe even unit tested).

At the very least we should change to not use the short-hand syntax.